mne/preprocessing/xdawn.py

```diff
@@ -1,266 +1,332 @@
+    else:
+        evokeds, toeplitzs = list(), list()
+        for c in classes:
+            # Prototyped responce for each class
+            evokeds.append(np.mean(epochs[y == c, :, :], axis=0))
+            toeplitzs.append(1.)
+
+    filters = []
+    patterns = []
+    for evo, toeplitz in zip(evokeds, toeplitzs):
+        # Covariance matrix of the prototyper response & signal
+        evo = np.dot(evo, toeplitz)
+        evo_cov = np.matrix(_regularized_covariance(evo, reg))
+
+        # Spatial filters
+        evals, evecs = linalg.eigh(evo_cov, signal_cov)
+        evecs = evecs[:, np.argsort(evals)[::-1]]  # sort eigenvectors
+        evecs /= np.apply_along_axis(np.linalg.norm, 0, evecs)
+        _patterns = np.linalg.pinv(evecs.T)
+        filters.append(evecs[:, :n_components].T)
+        patterns.append(_patterns[:, :n_components].T)
+
+    filters = np.concatenate(filters, axis=0)
+    patterns = np.concatenate(patterns, axis=0)
+    evokeds = np.array(evokeds)
+    return filters, patterns, evokeds
+
+
+class XdawnTransformer(BaseEstimator, TransformerMixin):
+    """Implementation of the Xdawn Algorithm.
 
-def _least_square_evoked_epochs(epochs_data, events, event_id, tmin,
-                                tmax, sfreq, return_toeplitz=False):
-    """Least square estimation of evoked response from a Epochs instance.
+    Xdawn is a spatial filtering method designed to improve the signal
+    to signal + noise ratio (SSNR) of the event related responses. Xdawn was
+    originaly designed for P300 evoked potential by enhancing the target
+    response with respect to the non-target response. This implementation is a
+    generalization to any type of event related response.
 
     Parameters
     ----------
-    epochs : Epochs instance
-        An instance of Epochs.
-    return_toeplitz : bool (default False)
-        If true, compute the toeplitz matrix.
-
-    Returns
-    -------
-    evokeds : dict of evoked instance
-        An dict of evoked instance for each event type in epochs.event_id.
-    toeplitz : dict of ndarray
-        If return_toeplitz is true, return the toeplitz matrix for each event
-        type in epochs.event_id.
-    """
-
-    events = events.copy()
-    events[:, 0] -= events[0, 0] + int(tmin * sfreq)
-    data = _construct_signal_from_epochs(events, sfreq, tmin, tmax,
-                                         epochs_data)
-    evoked_data, toeplitz = _least_square_evoked(data, events, event_id,
-                                                 tmin=tmin, tmax=tmax,
-                                                 sfreq=sfreq)
-    evokeds = dict()
-    for name, data in evoked_data.items():
-        evokeds[name] = data
-
-    if return_toeplitz:
-        return evokeds, toeplitz
-
-    return evokeds
+    n_components : int (default 2)
+        The number of components to decompose the signals.
+    reg : float | str | None (default None)
```

Comment:
It does, but @dengemann suggested it should be written in normal text and I changed it accordingly there. Or am I missing something here?